yarn-project/aztec.js/src/account/interface.ts

```diff
@@ -17,10 +19,23 @@ export type FeeOptions = {
 /** Creates authorization witnesses. */
 export interface AuthWitnessProvider {
   /**
-   * Create an authorization witness for the given message.
-   * @param message - Message to authorize.
+   * Computes an authentication witness from either a message or a caller and an action.
+   * If a message is provided, it will create a witness for the message directly.
+   * Otherwise, it will compute the message using the caller and the action.
+   * @param messageOrAuthWitInput - The message or the caller and action to approve
```

Comment:
I think the "AuthWitInput" in the param name is not that greate because I see as authwit input the input of this function. Maybe something like msgOrMsgPreimage would be better?